index.js

```diff
@@ -46,230 +48,230 @@ function resolveInputDirectory(inputNodes) {
+    let targetExtension;
+    if(!syntax)
+      syntax = 'scss';
+      this.linterConfig.syntax = syntax;
+    if(syntax === 'sugarss') {
+      targetExtension = 'sss';
+    } else {
+      targetExtension = syntax;
+    }
+    if(syntax === 'css'){
+      config.syntax = "";
+    }
+    extensions.push(targetExtension);
+    if(this.testPassingFiles || this.testFailingFiles)
+      targetExtension = 'stylelint-test.js';
+    this.extensions = extensions;
+    this.targetExtension = targetExtension;
+  }
 
-/**
- * Entry point for broccoli build
- * @override
- */
-StyleLinter.prototype.build = function() {
-  return Filter.prototype.build.call(this).finally(function() {
-  });
-};
+  /** Filter Class Overrides **/
 
-/**
- * This method is executed for every scss file, it:
- *  - Calls onError
- * @override
- */
-StyleLinter.prototype.processString = function(content, relativePath) {
- var self = this;
- this.linterConfig.code = content;
- this.linterConfig.codeFilename = path.join(this.inputNodesDirectory, relativePath);
- if(this.ignorer.ignores(this.linterConfig.codeFilename)){
-   return;
- }
- return stylelint.lint(this.linterConfig).then(function(results){
-  //sets the value to relative path otherwise it would be absolute path
-  results = self.processResults(results, relativePath);
-  if(results.errored && self.testFailingFiles) {
-   results.output = self.testGenerator(relativePath, results);
-  } else if(!results.errored && self.testPassingFiles) {
-     results.output = self.testGenerator(relativePath);
+  /**
+   * Entry point for broccoli build
+   * @override
+   */
+  build() {
+    return Filter.prototype.build.call(this).finally(function() {
+    });
   }
-  return results;
- }).catch(function(err) {
-   console.error(chalk.red('======= Something went wrong running stylelint ======='));
-   if(err.code === 78){
-     if(err.message.indexOf('No configuration provided') > -1){
-       console.error(chalk.red("No stylelint configuration found please create a .stylelintrc file in the route directory"));
-     } else {
-       console.error(chalk.red(err.message));
-     }
-   } else {
-     console.error(err.stack);
-   }
-   console.error(err.stack);
- });
-};
 
-/**
-  * @method postProcess
-  * This method is called after, the file has been linted:
-  *  - Logs to console
-  *  - Generate tests
-  * @override
-  */
-StyleLinter.prototype.postProcess = function(results, relativePath) {
-  if(results) {
-    if(results.errored){
-      if(this.onError) {
-       this.onError(results);
+  /**
+   * This method is executed for every scss file, it:
+   *  - Calls onError
+   * @override
+   */
+  processString(content, relativePath) {
+  let self = this;
+  this.linterConfig.code = content;
+  this.linterConfig.codeFilename = path.join(this.inputNodesDirectory, relativePath);
+  if(this.ignorer.ignores(this.linterConfig.codeFilename)){
+    return;
+  }
+  return stylelint.lint(this.linterConfig).then(function(results){
```

Comment:
we should be able to use a fat arrow here i think

Reply:
not sure if it is a v4 feature.

Reply:
done